src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,1133 @@
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+        });
+
+        VerifyOrReturn(triggerOptions.motionTimeControl.Value().maxDuration >= 1, {
+            ChipLogError(Zcl, "HandleAllocatePushTransport[ep=%d]: Motion Time Control (MaxDuration) Constraint Error",
+                         AttributeAccessInterface::GetEndpointId().Value());
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+        });
+    }
+
+    if (triggerOptions.triggerType == TransportTriggerTypeEnum::kMotion ||
+        triggerOptions.triggerType == TransportTriggerTypeEnum::kCommand)
+    {
+        VerifyOrReturn(triggerOptions.maxPreRollLen.HasValue(), {
+            ChipLogError(Zcl, "HandleAllocatePushTransport[ep=%d]: Missing Max Pre Roll Len field ",
+                         AttributeAccessInterface::GetEndpointId().Value());
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+        });
+    }
+
+    VerifyOrReturn(transportOptions.ingestMethod != IngestMethodsEnum::kUnknownEnumValue, {
+        ChipLogError(Zcl, "HandleAllocatePushTransport[ep=%d]: Invalid Ingest Method ",
+                     AttributeAccessInterface::GetEndpointId().Value());
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+    });
+
+    // Todo: TLSEndpointID Validation
+
+    IngestMethodsEnum ingestMethod = commandData.transportOptions.ingestMethod;
+
+    VerifyOrReturn(transportOptions.ingestMethod != IngestMethodsEnum::kUnknownEnumValue, {
+        ChipLogError(Zcl, "HandleAllocatePushTransport[ep=%d]: Invalid Ingest Method ",
+                     AttributeAccessInterface::GetEndpointId().Value());
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+    });
+
+    ContainerOptionsStruct containerOptions = commandData.transportOptions.containerOptions;
+
+    VerifyOrReturn(containerOptions.containerType != ContainerFormatEnum::kUnknownEnumValue, {
+        ChipLogError(Zcl, "HandleAllocatePushTransport[ep=%d]: Invalid Container Format ",
+                     AttributeAccessInterface::GetEndpointId().Value());
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+    });
+
+    if (containerOptions.containerType == ContainerFormatEnum::kCmaf)
+    {
+        VerifyOrReturn(containerOptions.CMAFContainerOptions.HasValue(), {
+            ChipLogError(Zcl, "HandleAllocatePushTransport[ep=%d]: Missing CMAF Container Options ",
+                         AttributeAccessInterface::GetEndpointId().Value());
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+        });
+
+        if (containerOptions.CMAFContainerOptions.Value().CENCKey.HasValue())
+        {
+            VerifyOrReturn(containerOptions.CMAFContainerOptions.Value().CENCKey.Value().size() <= kMaxCENCKeyLength, {
```

Comment:
```suggestion
            VerifyOrReturn(containerOptions.CMAFContainerOptions.Value().CENCKey.Value().size() == kMaxCENCKeyLength, {
```
Spec says length is exactly 16, not up to 16.

Reply:
corrected